semanticcpg/src/test/scala/io/shiftleft/semanticcpg/codedumper/CodeDumperTests.scala

```diff
@@ -0,0 +1,46 @@
+package io.shiftleft.semanticcpg.codedumper
+
+import io.shiftleft.semanticcpg.testfixtures.CodeToCpgFixture
+import org.scalatest.{Matchers, WordSpec}
+import io.shiftleft.semanticcpg.language._
+
+class CodeDumperTests extends WordSpec with Matchers {
+
+  val code = """
+                |int my_func(int param1) {
+                |   int x = foo(param1);
+                |}""".stripMargin
+
+  CodeToCpgFixture(code) { cpg =>
+    "should return empty string for empty traversal" in {
+      CodeDumper.dump(cpg.method.name("notinthere")) shouldBe ""
+    }
+
+    "should be able to dump complete function" in {
+      val query = cpg.method.name("my_func")
+      val code = CodeDumper.dump(query)
+      code should startWith(CodeDumper.arrow.toString)
+      code.contains("foo(param1)") shouldBe true
```

Comment:
You may want to test with `code should include(someString)` or `code should include regex """someRegex"""` for increased readability of match results.